src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -0,0 +1,136 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveDrivetrain;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveDrivetrainConstants;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveModule;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveModuleConstants;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveRequest;
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.commands.PathPlannerAuto;
+import com.pathplanner.lib.util.HolonomicPathFollowerConfig;
+import com.pathplanner.lib.util.PIDConstants;
+import com.pathplanner.lib.util.ReplanningConfig;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import frc.robot.Constants;
+import java.util.function.Supplier;
+
+/**
+ * Class that extends the Phoenix SwerveDrivetrain class and implements subsystem so it can be used
+ * in command-based projects easily.
+ */
+public class SwerveSubsystem extends SwerveDrivetrain implements Subsystem {
+  // instance of SwerveSubsystem
+  private static SwerveSubsystem instance;
+
+  // Constructor allows for custom odometry update frequency
+  public SwerveSubsystem(
+      SwerveDrivetrainConstants driveTrainConstants,
+      double OdometryUpdateFrequency,
+      SwerveModuleConstants... modules) {
+
+    // Sets the drivetrain constants, odometry update frequency and constants for
+    // the swerve modules
+    super(driveTrainConstants, OdometryUpdateFrequency, modules);
+
+    // Sets the supply current limits for the swerve modules (for driving and
+    // turning)
+    CurrentLimitsConfigs driveCurrentLimits =
+        new CurrentLimitsConfigs()
+            .withSupplyCurrentLimit(Constants.Swerve.kDriveSupplyCurrentLimit)
+            .withSupplyCurrentLimitEnable(true);
+
+    CurrentLimitsConfigs turningCurrentLimits =
```

Comment:
change to `steerCurrentLimits` for the same naming scheme as the constants

Reply:
done